### PR TITLE
Chet 141 parallel tests

### PR DESCRIPTION
**Pull request by theghostwhoforks**
Run tests in parallel using the github permutation matrix.

Notes -

1. Github does not support yaml anchors
2. Profiles are being use to set or unset integration test flags
3. 4 jobs are run in parallel instead of 2 sequential iterations before. While this does not reduce the time by half, it's still faster than before

### CODE REVIEW

File: .github/workflows/mvn_test.yml [hunk 1]
```diff
@@ -4,13 +4,13 @@ on: [push]
 
 jobs:
   run_tests:
-    name: Run tests with Java ${{ matrix.java }}
-
+    name: Java${{ matrix.java }} ${{ matrix.profile }} test
```
Comment (theghostwhoforks) [inline .github/workflows/mvn_test.yml]: Test name is rendered on screen as -
`Java8 integration test` and it fits in the tests pane on the left hand side of the actions page

**Comment by theghostwhoforks** [comment]
Ignore the commit list. Just one file has been changed in this PR

**Comment by theghostwhoforks** [comment]
I don't know how to get this merged in - GH says that the older tests that I've changed are required to merge this PR 🤷‍♂

File: .github/workflows/mvn_test.yml [hunk 2]
```diff
@@ -31,6 +31,4 @@ jobs:
         with:
           java-version: ${{ matrix.java }}
       - name: Test with maven
-        run: mvn -B verify
-      - name: Integration tests
-        run: mvn -B clean verify -Pintegration
+        run: mvn -B clean verify -P ${{ matrix.profile}}
```
Comment (theghostwhoforks) [inline .github/workflows/mvn_test.yml]: @bpar476 @nanux - Why do we need clean here? Won't it undo the effect of caching maven deps?

Comment (bpar476) [reply .github/workflows/mvn_test.yml]: I think it was just leftover from the template action. I can't think of any reason you'd want to clean.

**Pull request merged**